src/test/java/GoServletTest.java

```diff
@@ -0,0 +1,254 @@
+import com.google.sps.model.TasksClient;
+import com.google.sps.model.TasksClientFactory;
+import com.google.sps.servlets.GoServlet;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+@RunWith(JUnit4.class)
+public class GoServletTest extends AuthenticatedServletTestBase {
+
+  private DirectionsClientFactory directionsClientFactory;
+  private DirectionsClient directionsClient;
+  private PlacesClientFactory placesClientFactory;
+  private PlacesClient placesClient;
+  private TasksClientFactory tasksClientFactory;
+  private TasksClient tasksClient;
+  private GeocodingClientFactory geocodingClientFactory;
+  private GeocodingClient geocodingClient;
+  private GoServlet servlet;
+
+  private static final String API_KEY = "fake api key";
+  private static final String RESTAURANT_ONE = "restaurant near coordinate one";
+  private static final String RESTAURANT_TWO = "restaurant near coordinate two";
+  private static final String BANK_ONE = "bank near coordinate one";
+  private static final String BANK_TWO = "bank near coordinate two";
+
+  private GeocodingResult streetAddressGeocodingResult = new GeocodingResult();
+  private Geometry streetAddressGeometry = new Geometry();
+  private LatLng streetAddressCoordinates = new LatLng(0, 0);
+  private GeocodingResult restaurantGeocodingResult = new GeocodingResult();
+
+  private Duration longerDuration = new Duration();
+  private DirectionsLeg longerLeg = new DirectionsLeg();
+  private DirectionsRoute longerRoute = new DirectionsRoute();
+  private DirectionsResult longerResult = new DirectionsResult();
+
+  private Duration shorterDuration = new Duration();
+  private DirectionsLeg shorterLeg = new DirectionsLeg();
+  private DirectionsRoute shorterRoute = new DirectionsRoute();
+  private DirectionsResult shorterResult = new DirectionsResult();
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    directionsClientFactory = Mockito.mock(DirectionsClientFactory.class);
+    directionsClient = Mockito.mock(DirectionsClient.class);
+    placesClientFactory = Mockito.mock(PlacesClientFactory.class);
+    placesClient = Mockito.mock(PlacesClient.class);
+    tasksClientFactory = Mockito.mock(TasksClientFactory.class);
+    tasksClient = Mockito.mock(TasksClient.class);
+    geocodingClientFactory = Mockito.mock(GeocodingClientFactory.class);
+    geocodingClient = Mockito.mock(GeocodingClient.class);
+
+    streetAddressGeocodingResult.geometry = streetAddressGeometry;
+    streetAddressGeocodingResult.geometry.location = streetAddressCoordinates;
+    streetAddressGeocodingResult.types = new AddressType[] {AddressType.STREET_ADDRESS};
+    restaurantGeocodingResult.types = new AddressType[] {AddressType.RESTAURANT};
+
+    longerDuration.inSeconds = 10;
```

Comment:
Maybe set the time value to be a constant

Reply:
Good call, thanks!

Reply:
`RESOLVED`